### ISSUE TITLE
@typescript-eslint/no-use-before-define

### ISSUE THREAD

**Issue by GuoSirius** [typescript-eslint/typescript-eslint#2941]
```javascript
class A {
  constructor (printName) {
    this.printName = printName
  }

  openPort(printerName = this.printerName) {
    this.tscOcx.ActiveXopenport(printerName)

    return this
  }
}
```

![image](https://user-images.githubusercontent.com/48358523/104976037-355e3680-5a37-11eb-8a4c-19b37d4caf79.png)

![image](https://user-images.githubusercontent.com/48358523/104976055-4313bc00-5a37-11eb-91b8-d2972419bf00.png)

**Comment by GuoSirius** [comment]
```javascript
import { Vue, Component } from 'vue-property-decorator'

@Component({
  computed: {}
})
export default class NotFound extends Vue {}
```

![image](https://user-images.githubusercontent.com/48358523/104976380-0b594400-5a38-11eb-9391-b9dfeb5647f6.png)

**Comment by bradzacher** [comment]
The second issue is for the `no-undef` rule, not the `no-use-before-define` rule.

However - I am unable to repro either issue against master.

Please provide a repro repo which succinctly reproduces the error.

**Comment by bradzacher** [comment]
Same cause as #2942 - a regression introduced in a recent update in the scope analyser.

**Comment by paztis** [comment]
Facing this issue.
is it possible to release the fix in 4.14.1 ?
